slp2mp4/gui_test.py

```diff
@@ -0,0 +1,289 @@
+    video_output_var.set(video_output_path)
+
+# Create the main window
+root = tk.Tk()
+root.title("SLP2MP4 - Iowa Melee")
+
+# Variables to store user entries
+melee_iso_var = tk.StringVar()
+dolphin_directory_var = tk.StringVar()
+slippi_folder_var = tk.StringVar()
+selected_encoder = tk.StringVar()
+selected_resolution = tk.StringVar(value="1080p")  # Default value is set to 1080p
+selected_widescreen = tk.StringVar(value="False")  # Default value is set to False
+selected_combine = tk.StringVar()
+selected_remove_slps = tk.StringVar(value="False")  # Default value is set to False
+selected_remove_short_slps = tk.StringVar(value="False")  # Default value is set to False
+bitrate_var = tk.StringVar(value="20000")  # Default value is set to 20000
+
+#script and data folder path variables
+script_directory = os.path.dirname(os.path.realpath(__file__))
+data_folder = os.path.join(script_directory, "data")
+
+config_data = {
+    "melee_iso": melee_iso_var.get(),
+    "dolphin_dir": dolphin_directory_var.get(),
+    "resolution": selected_resolution.get(),
+    "video_backend": selected_encoder.get(),
+    "widescreen": selected_widescreen.get(),
+    "bitrateKbps": bitrate_var.get(),
+    "parallel_games": "recommended",
+    "remove_short": selected_remove_short_slps.get(),
+    "combine": selected_combine.get(),
+    "remove_slps": selected_remove_slps.get(),
+    "ffmpeg": "C:\\ffmpeg\\bin\\ffmpeg.exe"
+    # Can add other configuration parameters here as needed
+}
+
+
+
+def save_config():
+    config_data["melee_iso"] = melee_iso_var.get()
+    config_data["dolphin_dir"] = dolphin_directory_var.get()
+    config_data["resolution"] = selected_resolution.get()
+    config_data["video_backend"] = selected_encoder.get()
+    config_data["widescreen"] = selected_widescreen.get()
+    config_data["bitrateKbps"] = bitrate_var.get()
+    config_data["remove_short"] = selected_remove_short_slps.get()
+    config_data["combine"] = selected_combine.get()
+    config_data["remove_slps"] = selected_remove_slps.get()
+    # Add other config parameters here if needed
+
+    os.makedirs(data_folder, exist_ok=True)  # Ensure the "data" folder exists
+    config_file_path = Config(False).paths.config_json
+
+    with open(config_file_path, "w", encoding="utf-8") as config_file:
+        json.dump(config_data, config_file, indent=4)
+        config_file.truncate()
+
+#run slp2mp4 function
+def run_slp2mp4():
+    try:
+        script_directory = os.path.dirname(os.path.realpath(__file__))
+        slp2mp4_script = os.path.abspath(os.path.join(script_directory, "slp2mp4.py"))
+        slippi_folder_var.get() #maybe not necessary in python
+        print([slp2mp4_script, "run", video_output_var.get(), slippi_folder_var.get() ])
+        subprocess.run(["python", slp2mp4_script, "run", video_output_var.get(), slippi_folder_var.get() ], check=True, shell=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error running slp2mp4: {e}")
```

Comment:
Instead of using `subprocess`, I'd prefer this use the same mechanism as `run()` in `slp2mp4.py`, that way it's consistent